### PR TITLE
Handle unsupported platform errors gracefully

### PR DESCRIPTION
**Pull request by gnufied**
Unsupported platform isn't really a fatal error. We are just not going to install a storageclass. This fixes deployment on libvirt.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Yeah, as a general rule all operators everywhere must gracefully tolerate unrecognized platforms.  See https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L34 for a quick description and 
https://github.com/openshift/installer/pull/1112 for more details

**Comment by jsafrane** [comment]
This code lgtm, however, should the operator tell user somehow that it's running correctly but did not do anything useful on an unknown platform? *"I see the operator running, it does not report any error, so where is my default storage class?"*

File: pkg/controller/clusterstorage/clusterstorage_controller.go [hunk 1]
```diff
@@ -192,7 +193,8 @@ func (r *ReconcileClusterStorage) syncStatus(clusterOperator *configv1.ClusterOp
 	}
 	v1helpers.SetStatusCondition(&clusterOperator.Status.Conditions, notProgressing)
 
-	if err != nil {
+	// if error is anything other than unsupported platform, we are failing
+	if err != nil && err != unsupportedPlatformError {
```
Comment (bertinatto) [inline pkg/controller/clusterstorage/clusterstorage_controller.go]: +1: it could be useful to have a log entry if `err == unsupportedPlatformError`.

Comment (gnufied) [reply pkg/controller/clusterstorage/clusterstorage_controller.go]: fixeds

**Comment by gnufied** [comment]
@jsafrane added reason field if unsupported platform is detected. PTAL

**Comment by wongma7** [comment]
reason should be camelcase. maybe put it in Message instead?

**Comment by gnufied** [comment]
@wongma7 fixed.

**Pull request merged**